src/databricks/labs/remorph/snow/sql_transpiler.py

```diff
@@ -1,26 +1,46 @@
+import logging
+import typing as t
 from sqlglot import expressions as exp, parse, transpile
 from sqlglot.dialects.dialect import Dialect
 from sqlglot.errors import ErrorLevel, ParseError, TokenError, UnsupportedError
 from sqlglot.expressions import Expression
+from sqlglot.tokens import Token, TokenType
 
 from databricks.labs.remorph.config import TranspilationResult
-from databricks.labs.remorph.helpers.file_utils import refactor_hexadecimal_chars
+from databricks.labs.remorph.helpers.file_utils import refactor_hexadecimal_chars, format_error_message
 from databricks.labs.remorph.helpers.morph_status import ParserError
 
+logger = logging.getLogger(__name__)
+
 
 class SqlglotEngine:
     def __init__(self, read_dialect: Dialect):
         self.read_dialect = read_dialect
 
+    def partial_transpile(
+        self, write_dialect: Dialect, sql: str, file_name: str, error_list: list[ParserError]
+    ) -> tuple[list[str], list[ParserError]]:
+        transpiled_sql_statements = []
+        parsed_expressions, errors = self.safe_parse(statements=sql, read=self.read_dialect)
+        for expression in parsed_expressions:
+            if expression is not None:
+                transpiled_sql = write_dialect.generate(expression, pretty=True)
+                transpiled_sql_statements.append(transpiled_sql)
+        for error in errors:
```

Comment:
I will move this to a private function. just make a function call

```suggestion
        _handle_errors(errors)
```